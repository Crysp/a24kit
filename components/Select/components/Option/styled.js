import styled from 'styled-components';
import Checkbox from '../../../Checkbox';
import { Color } from '../../../config';

export const Wrapper = styled.div`
    padding: 6px 12px;
    font: 400 14px "Open Sans", Helvetica, Arial, sans-serif;
    line-height: 20px;
    cursor: pointer;
    background-color: ${props => (props.selected ? '#e4edf7' : Color.White)};
    transition: background-color .12s linear;
    &:hover {
        background-color: ${Color.Zircon};
    }
`;

export const StyledCheckbox = styled(Checkbox)`
    width: 100%;
`;
