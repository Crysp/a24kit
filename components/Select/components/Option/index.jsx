import React from 'react';
import PropTypes from 'prop-types';
import { Wrapper, StyledCheckbox } from './styled';


export default class Option extends React.PureComponent {
    static propTypes = {
        label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
        hasCheckbox: PropTypes.bool,
        isChecked: PropTypes.bool,
        isSelected: PropTypes.bool,
        onSelect: PropTypes.func,
    };

    static defaultProps = {
        hasCheckbox: false,
        isChecked: false,
        isSelected: false,
        onSelect() {},
    };

    render() {
        const {
            label,
            hasCheckbox,
            isChecked,
            isSelected,
            onSelect,
        } = this.props;
        return (
            <Wrapper selected={isSelected} onClick={onSelect}>
                {hasCheckbox ? <StyledCheckbox label={label} isChecked={isChecked} /> : label}
            </Wrapper>
        );
    }
}
