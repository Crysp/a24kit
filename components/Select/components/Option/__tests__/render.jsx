import React from 'react';
import { create } from 'react-test-renderer';
import Option from '../index';


describe('Верстка компонента <Select.Option>', () => {
    it('Дефолтная', () => {
        const tree = create(<Option label="Label" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Выбранный', () => {
        const tree = create(<Option label="Label" isSelected />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('С чекбоксом', () => {
        const tree = create(<Option label="Label" hasCheckbox />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('С выбранным чекбоксом', () => {
        const tree = create(<Option label="Label" hasCheckbox isChecked />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
