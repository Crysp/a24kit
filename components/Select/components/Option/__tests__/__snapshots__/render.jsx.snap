// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Верстка компонента <Select.Option> Выбранный 1`] = `
<div
  className="sc-bZQynM heCUJq"
  onClick={[Function]}
  selected={true}
>
  Label
</div>
`;

exports[`Верстка компонента <Select.Option> Дефолтная 1`] = `
<div
  className="sc-bZQynM cCCPgu"
  onClick={[Function]}
  selected={false}
>
  Label
</div>
`;

exports[`Верстка компонента <Select.Option> С выбранным чекбоксом 1`] = `
<div
  className="sc-bZQynM cCCPgu"
  onClick={[Function]}
  selected={false}
>
  <div
    className="sc-gzVnrw fUdNQC sc-ifAKCX gkzOrl"
    color="blue"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH dgURRx"
      disabled={false}
    >
      <div
        className="sc-htpNat cawDMS"
        color="blue"
        disabled={false}
      >
        <svg
          aria-hidden="true"
          className="sc-bxivhb himSXy sc-bdVaJa dTKkTG"
          focusable="false"
          size="auto"
          style={Object {}}
          viewBox="0 0 8 7"
        >
          <path
            d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
          />
        </svg>
      </div>
    </div>
    <div
      className="sc-EHOje hijERh"
      disabled={false}
    >
      Label
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <Select.Option> С чекбоксом 1`] = `
<div
  className="sc-bZQynM cCCPgu"
  onClick={[Function]}
  selected={false}
>
  <div
    className="sc-gzVnrw fUdNQC sc-ifAKCX gkzOrl"
    color="blue"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH dSdhvk"
      disabled={false}
    >
      <div
        className="sc-htpNat bqsoxB"
        color="blue"
        disabled={false}
      >
        <svg
          aria-hidden="true"
          className="sc-bxivhb himSXy sc-bdVaJa dTKkTG"
          focusable="false"
          size="auto"
          style={Object {}}
          viewBox="0 0 8 7"
        >
          <path
            d="M2.59 7a.74.74 0 0 1-.5-.2L.21 4.95a.68.68 0 0 1 0-.98.7.7 0 0 1 .99 0l1.39 1.36L6.8 1.2a.7.7 0 0 1 .99 0c.28.27.28.71 0 .98L3.09 6.8a.74.74 0 0 1-.5.2"
          />
        </svg>
      </div>
    </div>
    <div
      className="sc-EHOje hijERh"
      disabled={false}
    >
      Label
    </div>
  </div>
</div>
`;
